MapboxCoreNavigation/RouteController.swift

```diff
@@ -334,19 +392,22 @@ extension RouteController: CLLocationManagerDelegate {
         updateVisualInstructionProgress()
 
         guard userIsOnRoute(location) || !(delegate?.routeController?(self, shouldRerouteFrom: location) ?? true) else {
-            reroute(from: location, along: routeProgress)
+            rerouteForDiversion(from: location, along: routeProgress)
             return
         }
 
         updateSpokenInstructionProgress()
 
         // Check for faster route given users current location
         guard reroutesProactively else { return }
-        // Only check for faster alternatives if the user has plenty of time left on the route.
+        
+        // Only check for faster routes or ETA updates if the user has plenty of time left on the route (10+min)
         guard routeProgress.durationRemaining > 600 else { return }
```

Comment:
Why wouldn't we check for ETA updates in the last 10 minutes. I get you don't want to reroute anymore, but updated ETA is still relevant.

Reply:
Backend was strict about not supporting more calls than we currently do client-side. We could configure this in the SDK so other clients can skip this check? What do you think?

Reply:
If it is a client requirement, we should make it configurable by the client